packages/components/src/font-size-picker/index.js

```diff
@@ -6,51 +6,108 @@ import { map } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { withInstanceId } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
+import Dashicon from '../dashicon';
+import BaseControl from '../base-control';
 import Button from '../button';
-import ButtonGroup from '../button-group';
+import Dropdown from '../dropdown';
 import RangeControl from '../range-control';
+import { NavigableMenu } from '../navigable-container';
+
+function FontSizePicker( {
+	fallbackFontSize,
+	fontSizes = [],
+	instanceId,
+	onChange,
+	value,
+	withSlider,
+} ) {
+	const onChangeValue = ( event ) => {
+		const newValue = event.target.value;
+		if ( newValue === '' ) {
+			onChange( undefined );
+			return;
+		}
+		onChange( Number( newValue ) );
+	};
+
+	const currentFont = fontSizes.find( ( font ) => font.size === value );
+	const labelId = `components-font-size-picker-label-${ instanceId }`;
 
-export default function FontSizePicker( { fontSizes = [], fallbackFontSize, value, onChange } ) {
 	return (
-		<Fragment>
+		<BaseControl
+			label={ __( 'Font Size' ) }
+			id={ labelId }
+		>
 			<div className="components-font-size-picker__buttons">
-				<ButtonGroup aria-label={ __( 'Font Size' ) }>
-					{ map( fontSizes, ( { name, size, shortName } ) => (
-						<Button
-							key={ size }
-							isLarge
-							isPrimary={ value === size }
-							aria-pressed={ value === size }
-							onClick={ () => onChange( size ) }
-						>
-							{ shortName || name }
+				<Dropdown
+					className="components-font-size-picker__dropdown"
+					contentClassName="components-font-size-picker__dropdown-content"
+					position="bottom"
+					renderToggle={ ( { isOpen, onToggle } ) => (
+						<Button className="components-font-size-picker__selector" isLarge onClick={ onToggle } aria-expanded={ isOpen }>
+							{ ( currentFont && currentFont.name ) || ( ! value && 'Normal' ) || 'Custom' }
 						</Button>
-					) ) }
-				</ButtonGroup>
+					) }
+					renderContent={ () => (
+						<NavigableMenu aria-labelledby={ labelId }>
+							{ map( fontSizes, ( { name, size, slug } ) => (
+								<Button
+									key={ slug }
+									aria-pressed={ value === size }
+									onClick={ () => onChange( slug === 'normal' ? undefined : size ) }
+									className={ 'is-font-' + slug }
+								>
+									{ ( value === size || ( ! value && slug === 'normal' ) ) &&	<Dashicon icon="saved" /> }
+									<span className="components-font-size-picker__dropdown-text-size" style={ { fontSize: size } }>
+										{ name }
+									</span>
+								</Button>
+							) ) }
+						</NavigableMenu>
+					) }
+				/>
+				{ ! withSlider &&
+					<input
+						className="components-range-control__number"
+						type="number"
+						onChange={ onChangeValue }
+						aria-label={ __( 'Custom Size' ) }
```

Comment:
I'd suggest to clarify and use "custom font size". Also, since this is invisible to the eyes, it doesn't need to be title cased.